packages/ecmascript/src/Map.ts

```diff
@@ -56,26 +56,23 @@ export default class Map<K extends (string | number), V> {
 		return values;
 	}
 
-	has(key): boolean {
+	public has(key: any): boolean {
 		return this.items.hasOwnProperty(key);
 	}
 
-	get(key): any {
+	public get(key: any): any {
 		return this.items[key];
 	}
 
-	set(key, value): any {
-		let exist = this.items.hasOwnProperty(key);
-		if (exist) {
-			this.size++;
-		}
-
+	public set(key: any, value: any): any {
 		this.items[key] = value;
+		this.size++;
+
 		return this;
 	}
```

Comment:
Still need to check if the key already exists.
Otherwise the size will always increment even when overwriting an existing key.

Reply:
You're right, the code was updated accordingly, thanks for the note.